### ISSUE TITLE
One line lambda function return cause 'Unexpected use of comma operator. (no-sequences)' error

### ISSUE THREAD

**Issue by Thaina** [eslint/eslint#7827]
I write code like this

```js
var s = ["TT","TTTT"].reduce((obj,s,i) => (obj[s] = i,obj),{});
```

And it was allowed as ES2015 syntax to return `obj` after set `obj[s] = i`
But eslint tell me its not

Why?

**Comment by platinumazure** [comment]
You have a couple of options: Explicitly return, or disable the rule for that line. 

```js
// Explicit return

var s = ["TT","TTTT"].reduce((obj,s,i) => { obj[s] = i; return obj; }, {});

// Disable the rule 

// eslint-disable-next-line no-sequences
var s = ["TT","TTTT"].reduce((obj,s,i) => (obj[s] = i,obj),{});
```

**Comment by Thaina** [comment]
@platinumazure Well, I know about option. I just want to know the reason that why this is consider error in eslint. Is it a bad practice or it just not update yet?

**Comment by platinumazure** [comment]
It's because you're using a sequence and the rule is designed to flag
against sequences. The rule is not necessarily claiming that all sequences
are bad practice; but even if it were, we still have disable comments for
this exact reason, when the utility in one given case outweighs the normal
need for the rule.
